user_group_popover: Redesign popover using the new "popover-menu" theme.

sayamsamal opened the pull request:
As part of the popover menu redesign, this PR redesigns the user group info popover using the new "popover-menu" tippy theme and improves accessibility by using appropriate ARIA attributes.

Fixes part of #28699.

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**

| Header | Header | Header |
|--------|--------|--------|
| Light Mode | <img width="480" alt="image" src="https://github.com/zulip/zulip/assets/82862779/37d87228-81ab-4323-a78f-34b8fa920ee5"> | <img width="480" alt="image" src="https://github.com/zulip/zulip/assets/82862779/4c556906-2337-4b56-af4f-451f59eae7c7"> |
| Dark Mode | <img width="480" alt="image" src="https://github.com/zulip/zulip/assets/82862779/0ec1c505-53eb-41b6-a169-00945aff1d70"> | <img width="480" alt="image" src="https://github.com/zulip/zulip/assets/82862779/940e9a49-eed8-426c-9aa3-bcbe2d825cde"> |
| Members list long enough to scroll | <img width="480" alt="image" src="https://github.com/zulip/zulip/assets/82862779/8182098e-693d-497e-a672-e3e9c8a1c4e1"> | <img width="480" alt="image" src="https://github.com/zulip/zulip/assets/82862779/fbc4539a-49d1-47cd-94d3-a5d5a52f8482"> | 

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

timabbott commented on web/styles/popovers.css:
```diff
@@ -352,63 +352,56 @@ ul {
-            text-align: center;
-        }
-    }
+.popover-menu-member-name {
```
`popover-group-menu` would be a better prefix than `popover-menu` for these, just to be consistent with the rest and clearly scoped to the group popover.

sayamsamal replied:
Does sound better; Replaced the classes with the `popover-group-menu` prefix, thanks!

alya commented:
Hm, the keyboard nav doesn't seem to be working for me here. When I press the down arrow, the settings link get highlighted. But nothing happens when I then press `Enter`.

alya commented:
Looks good to me in manual testing other than that!

alya commented:
Oh, one more: we should set the appropriate margin for user names in the logged out UI, which doesn't show presence status:

![Screenshot 2024-07-03 at 16 38 35@2x](https://github.com/zulip/zulip/assets/2090066/5af9d981-ca3e-43b7-9d92-81b490231c30)

sayamsamal commented:
> Hm, the keyboard nav doesn't seem to be working for me here. When I press the down arrow, the settings link gets highlighted. But nothing happens when I then press `Enter`.

Nice catch, fixed, thanks!

sayamsamal commented:
> Oh, one more: we should set the appropriate margin for user names in the logged out UI, which doesn't show presence status:

I added a `10px` padding on either side, so now it looks uniform in the spectators' view:
<img width="480" alt="image" src="https://github.com/zulip/zulip/assets/82862779/d75e50fb-8bb6-43aa-a902-395461b4825a">

Although, that shifts the member list to a little right in the normal view, does that work for you?
<img width="480" alt="image" src="https://github.com/zulip/zulip/assets/82862779/214b7bb0-46af-4f1a-bf1a-58f1be561e28">

sayamsamal commented:
## Updates:
- Changed the group info popover-related class names to use the `popover-group-menu` prefix
- Fixed padding in spectators view, where the user presence icon is not shown.
- Fixed `Enter` key not opening **Group Settings**.
- Removed **Group Settings** menu option for **Guests** and **Spectators**.

sayamsamal commented:
@alya can you please review this again? Thanks!

alya commented:
Huh, something went awry with light theme:

![Screenshot 2024-07-05 at 17 04 26@2x](https://github.com/zulip/zulip/assets/2090066/3d05f307-9718-4e94-b7a6-6ff4b00beb26)

Dark theme looks fine.

alya commented:
Actually, the font colors look off in dark theme as well.

alya commented:
Looks good to me in manual testing other than that.

sayamsamal commented:
@alya I have fixed the font color for the member list and the group description.

<img width="354" alt="image" src="https://github.com/zulip/zulip/assets/82862779/0181e597-174f-4e91-8194-3f7e713e40ae">
<img width="354" alt="image" src="https://github.com/zulip/zulip/assets/82862779/b325afbb-d22e-474e-b36a-95e2fd6ec3fd">

alya commented:
Ok, @timabbott do you want to review now? I haven't re-tested.

timabbott commented:
Merged, thanks @sayamsamal!
